Dockerfile

```diff
@@ -1,5 +1,119 @@
-ARG image_tag=latest
-FROM elifesciences/pattern-library_ui-builder:${image_tag} AS ui-builder
+ARG node_version
+ARG environment=production
+ARG description=unknown
+ARG selenium_image_suffix=
+
+# NPM - Manages the node part of the app
+
+FROM node:${node_version} as npm
+
+COPY npm-shrinkwrap.json \
+    package.json \
+    ./
+
+RUN npm install
+
+# Assets-Builder - Builds the assets (css, js, etc)
+
+FROM npm as assets_builder
+
+WORKDIR /srv/pattern-library
+
+COPY .babelrc \
+    .jscsrc \
+    .jshintrc \
+    .stylelintrc \
+    gulpfile.js \
+    ./
+
+COPY --from=npm /node_modules/ node_modules/
+
+ARG environment
+
+COPY assets/fonts/ assets/fonts/
+RUN node_modules/.bin/gulp --environment ${environment} fonts
+
+COPY assets/img/ assets/img/
+RUN node_modules/.bin/gulp --environment ${environment} img
+
+COPY assets/js/ assets/js/
+RUN node_modules/.bin/gulp --environment ${environment} js
+
+COPY assets/sass/ assets/sass/
+RUN node_modules/.bin/gulp --environment ${environment} generateCss
+
+# Assets
+
+FROM assets_builder
```

Comment:
Lines that take a FROM, but don't add any other layers or commands, don't actually do anything and can be removed